app/src/main/java/com/fstyle/androidtrainning/service/ServicePlayMusic.java

```diff
@@ -35,22 +41,36 @@ public ServicePlayMusic() {
 
     @Override
     public IBinder onBind(Intent intent) {
-        Track track = intent.getParcelableExtra(Constant.EXTRA_TRACK_ITEM);
-        id = track.getPosition();
-        playMusic();
         return mIBinder;
     }
 
     @Override
     public void onCreate() {
         super.onCreate();
-        mExternalData.scanAllMusic(this);
-        mExternalData.scanAllAlbum(this);
-        mExternalData.scanAllArtist(this);
+        mMediaPlayer = new MediaPlayer();
     }
 
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
+        if (intent != null) {
+            if (intent.getParcelableArrayListExtra(Constant.EXTRA_TRACK_LIST_ITEM) != null) {
```

Comment:
WARNING: [CollapsibleIfStatements] These nested if statements could be combined
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#CollapsibleIfStatements